src/test/java/org/takes/http/FtBasicTest.java

```diff
@@ -196,4 +204,33 @@ public void exec(final URI home) throws IOException {
         );
     }
 
+    /**
+     * FtBasic can consume twice the input stream.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void consumesTwiceInputStream() throws IOException {
```

Comment:
@krzyk What about assertStatus and assertBody that you have at each call? it is not enough?